tests/pine.spec.coffee

```diff
@@ -60,68 +74,129 @@ describe 'Pine:', ->
 
 				describe 'given a simple GET endpoint', ->
 
-					beforeEach ->
-						@pine = buildPineInstance(mockServer.url)
-						mockServer.get('/foo').thenJSON(200, hello: 'world')
+					describe 'given a public resource', ->
+
+						beforeEach ->
+							@pine = buildPineInstance(mockServer.url)
+							mockServer.get('/public_resource').thenJSON(200, hello: 'public world')
+
+						describe 'given there is no api key', ->
+
+							it 'should be successful', ->
+								promise = @pine._request
+									baseUrl: @pine.API_URL
+									method: 'GET'
+									url: '/public_resource'
+								m.chai.expect(promise).to.be.rejectedWith('You have to log in')
+
+							it 'should be successful, if sent anonymously', ->
+								promise = @pine._request
+									baseUrl: @pine.API_URL
+									method: 'GET'
+									url: '/public_resource'
+									anonymous: true
+								m.chai.expect(promise).to.become(hello: 'public world')
+
+						describe 'given there is an api key', ->
+
+							it 'should make the request successfully', ->
+								promise = @pine._request
+									baseUrl: @pine.API_URL
+									method: 'GET'
+									url: '/public_resource'
+								m.chai.expect(promise).to.be.rejectedWith('You have to log in')
+
+					describe 'given a non-public resource', ->
+
+						describe 'given there is no api key', ->
+
+							beforeEach ->
+								@pine = buildPineInstance(mockServer.url, apiKey: '')
+
+							it 'should be rejected with an authentication error message', ->
+								promise = @pine._request
+									baseUrl: @pine.API_URL
+									method: 'GET'
+									url: '/foo'
+								m.chai.expect(promise).to.be.rejectedWith('You have to log in')
+
+							it 'should be rejected with an unauthorized error, if sent anonymously', ->
+								promise = @pine._request
+									baseUrl: @pine.API_URL
+									method: 'GET'
+									url: '/foo'
+									anonymous: true
+								m.chai.expect(promise).to.be.rejectedWith('Unauthorized')
+								.then (res) ->
+									m.chai.expect(res.statusCode).to.equal(401)
```

Comment:
This was a bug in the previous tests, since it should reject requests w/o apiKey and token on non-public resources.